src/components/hand-controls.js

```diff
@@ -74,34 +126,95 @@ module.exports.Component = registerComponent('hand-controls', {
    * @param {string} evt the event associated to the button
    */
   handleButton: function (button, evt) {
-    var el = this.el;
     var isPressed = evt === 'down';
-    switch (button) {
-      case 'trackpad':
-        if (isPressed === this.trackpadPressed) { return; }
-        this.trackpadPressed = isPressed;
-        this.playAnimation('thumb', !isPressed);
-        evt = isPressed ? 'thumbup' : 'thumbdown';
-        el.emit(evt);
-        break;
-      case 'trigger':
-        if (isPressed === this.triggerPressed) { return; }
-        this.triggerPressed = isPressed;
-        this.playAnimation('pointing', !isPressed);
-        evt = isPressed ? 'pointup' : 'pointdown';
-        el.emit(evt);
-        break;
-      case 'grip':
-        if (isPressed === this.gripPressed) { return; }
-        this.gripPressed = isPressed;
-        this.playAnimation('close', !isPressed);
-        evt = isPressed ? 'gripclose' : 'gripopen';
-        el.emit(evt);
-        break;
+    var isTouched = evt === 'touchstart';
+    if (evt.indexOf('touch') === 0) {
+      if (isTouched === this.touchedButtons[button]) { return; }
+      this.touchedButtons[button] = isTouched;
+    } else {
+      if (isPressed === this.pressedButtons[button]) { return; }
+      this.pressedButtons[button] = isPressed;
+    }
+    if (!this.determineGesture()) { return; }
+    this.animateGesture();
+    this.emitGestureEvents();
+  },
+
+  determineGesture: function () {
+    var lastGesture = this.gesture;
+    var gesture;
+    if (this.pressedButtons['grip']) {
+      if (this.pressedButtons['surface'] || this.touchedButtons['surface'] ||
```

Comment:
this `if` conditions are hard to read and understand. We can clean up by making variables on the top:
````javascript
var isSurfaceButtonActive = this.pressedButtons['surface'] || this.touchedButtons['surface'];
var isTrackpadActive = this.pressedButtons['trackpad'] || this.touchedButtons['trackpad'];
var isTriggerActive = this.pressedButtons['trigger'] || this.touchedButtons['trigger'];
var isGripActive = this.pressedButtons['grip'];
var isAorXActive = this.pressedButtons['AorX'] || this.touchedButtons['AorX']
if (isGripActive) {
 if (isSurfaceActive || isTrackpadActive || isTriggerActive || isAorXActive) {
 if (isTriggerActive} { 
 ...
 } else {... }
}
...
`````
}